rewrite-java-core/src/test/java/ExceptionTypesTest.java

```diff
@@ -0,0 +1,148 @@
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.Test;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+public class ExceptionTypesTest implements RewriteTest {
+    /**
+     * ExceptionTypesTest tests exception migrations from azure-core v1
+     * to azure-core-v2 and client-core.
+     * Recipes used: ChangeType
+     * From:
+     * com.azure.core.exception
+     *      ClientAuthenticationException
+     *      HttpResponseException
+     *      ResourceModifiedException
+     *      ResourceNotFoundException
+     * To:
+     * com.azure.core.v2.exception
+     *      ClientAuthenticationException
+     *      ResourceModifiedException
+     *      ResourceNotFoundException
+     * io.clientcore.core.http.exception
+     *      HttpResponseException
+     *
+     * @author Annabelle Mittendorf Smith
+     */
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipeFromResource("/META-INF/rewrite/rewrite.yml",
+                "com.azure.rewrite.java.core.MigrateAzureCoreSamplesToAzureCoreV2");
+    }
+
+    /* Testing ChangeType recipes */
+    @Test
+    public void testClientAuthenticationExceptionChanged() {
+        @Language("java") String before = "import com.azure.core.exception.ClientAuthenticationException;";
+        before += "\npublic class Testing {";
+        before += "\n  public Testing(){";
+        before += "\n    com.azure.core.exception.ClientAuthenticationException e = new ClientAuthenticationException(null,null);";
+        before += "\n  }";
+        before += "\n}";
+
+        @Language("java") String after = "import com.azure.core.v2.exception.ClientAuthenticationException;";
+        after += "\n\npublic class Testing {";
+        after += "\n  public Testing(){";
+        after += "\n    com.azure.core.v2.exception.ClientAuthenticationException e = new ClientAuthenticationException(null,null);";
```

Comment:
Users generally don't new up exception types. They just handle these exceptions in a try-catch block. So, we should have tests for converting the exceptions in `catch` block.

Reply:
Modified exception tests to test out exceptions in catch blocks in commit [492daeb](https://github.com/Azure/azsdk-java-rewrite-recipes/pull/14/commits/492daeb48360470829cc54218911b9b3e1758ace)